### PR TITLE
Add validation for entity fetcher

### PR DESCRIPTION
**Pull request by kmoore-intuit**
# What Changed

# Why

Todo:

- [ ] Add tests
- [ ] Add docs

### CODE REVIEW

File: src/main/java/com/intuit/graphql/orchestrator/federation/EntityDataFetcher.java [hunk 1]
```diff
@@ -85,10 +86,22 @@ public CompletableFuture<Object> get(final DataFetchingEnvironment dataFetchingE
             result -> {
               // TODO transformer
               Map<String, Object> data = (Map<String, Object>) result.get(DATA_RESPONSE_FIELD);
-              List<Map<String, Object>> _entities =
-                  (List<Map<String, Object>>) data.get(_ENTITIES_FIELD_NAME);
+              List<Map<String, Object>> _entities = (data != null) ? (List<Map<String, Object>>) data.get(_ENTITIES_FIELD_NAME) : null;
```
Comment (bhavinshah7) [inline src/main/java/com/intuit/graphql/orchestrator/federation/EntityDataFetcher.java:89]: Is this null check needed? wouldn't the previous line result in null pointer while casting to Map  if data is null?

**Review by bhavinshah7 — APPROVED** [review]

File: src/test/java/com/intuit/graphql/orchestrator/federation/EntityDataFetcherTest.java [hunk 2]
```diff
@@ -181,6 +184,131 @@ public void get_no_entity_field_exception() throws ExecutionException, Interrupt
     }
   }
 
+  @Test
+  public void get_null_entity_field_exception() throws ExecutionException, InterruptedException {
```
Comment (CNAChino) [inline src/test/java/com/intuit/graphql/orchestrator/federation/EntityDataFetcherTest.java:188]: Might want to add  integration test where EntityDataFetcher get an `error` and empty data.

**Review by CNAChino — APPROVED** [review]

File: src/test/java/com/intuit/graphql/orchestrator/federation/EntityDataFetcherTest.java (hunk 2 above)
Comment (kmoore-intuit) [reply src/test/java/com/intuit/graphql/orchestrator/federation/EntityDataFetcherTest.java:188]: I'll fix that in my next commit

File: src/main/java/com/intuit/graphql/orchestrator/federation/EntityDataFetcher.java (hunk 1 above)
Comment (kmoore-intuit) [reply src/main/java/com/intuit/graphql/orchestrator/federation/EntityDataFetcher.java:89]: Nope; that would just remain as null. I will do the error check first, so it doesn't do an unnecessary work.
